rspace/src/test/scala/coop/rchain/rspace/history/HistorySpec.scala

```diff
@@ -100,22 +239,45 @@ class HistorySpec extends FlatSpec with Matchers with OptionValues with InMemory
       case Skip(affix, _) => affix.toSeq.toList shouldBe bytes
       case p              => fail("unknown trie prefix" + p)
     }
+
+  def randomKey(size: Int) = List.fill(size)((Random.nextInt(256) - 128).toByte)
+
+  def generateRandomInsert(size: Int) = List.fill(size)(insert(randomKey(32)))
+  def generateRandomDelete(size: Int) = List.fill(size)(delete(randomKey(32)))
+  def generateRandomDeleteFromInsert(size: Int, inserts: List[InsertAction]) =
+    Random.shuffle(inserts).take(size).map(i => delete(i.key))
+
+  def updateState(state: TrieMap[KeyPath, Blake2b256Hash], actions: List[HistoryAction]) = {
+    actions.map(
+      a =>
+        a match {
+          case InsertAction(key, value) => state.put(key, value)
+          case DeleteAction(key)        => state.remove(key)
+        }
+    )
+    state
+  }
+
 }
 
 object TestData {
 
   implicit def toByteVector(bytes: KeyPath): ByteVector = ByteVector(bytes)
 
-  val _zeros: KeyPath           = List.fill(32)(0).map(_.toByte)
-  val _zerosOnes: KeyPath       = (List.fill(16)(0) ++ List.fill(16)(1)).map(_.toByte)
-  val _31zeros: KeyPath         = List.fill(31)(0).map(_.toByte)
+  val _zeros: KeyPath = hexKey("0000000000000000000000000000000000000000000000000000000000000000")
+  val _zerosOnes: KeyPath = hexKey(
+    "0000000000000000000000000000000001010101010101010101010101010101"
+  )
+  val _31zeros: KeyPath         = hexKey("00000000000000000000000000000000000000000000000000000000000000")
```

Comment:
In this case seems more readable definition with _List.fill_ where numbers of bytes can be easily compared and with hex is difficult to validate except counting digits, especially now when they are not aligned.